### DISCUSSION TITLE
Support for file extension in route/URL like /humans.txt

### DISCUSSION THREAD

**Discussion by pke** [remix-run/remix#8718] [Q&A] [answered]
Migrating from nextjs I'd like to keep a [humans.txt](https://humanstxt.org/) around my website root.
This file needs to be dynamically created (during build time containing the build time) and served as a static resource with cache headers.

I read trough the docs but all I see are routes without extensions. How would one create such route and keep the `.txt` extension and not just end up with `/humans` as route?

**Comment by thinkafterbefore** [comment, marked as answer]
Simply put the file extension in brackets like this `robots[.txt].ts`

**Reply by pke** [reply]
Did I miss this in the docs somehow or is it not in the docs?
And this could also be a `.mdx` file I assume?

**Reply by alexanderson1993** [reply]
It's referenced here: https://remix.run/docs/en/main/file-conventions/routes#escaping-special-characters
And here: https://remix.run/docs/en/main/guides/resource-routes#url-escaping

**Reply by pke** [reply]
Blind me! Thanks!

**Reply by alexanderson1993** [reply]
It's easy to miss. Not sure how to solve for it - sometimes you just have to read docs from top to bottom to get an inkling of where certain topics are mentioned.

**Reply by pke** [reply]
may the paragraph could contain something like a clearly visible example. They are all hidden in the code section and easy to muss. A bold route like this?

> You'll probably want to add a file extension to your resource routes like **`/sitemap.xml`** or **`/robots.txt`**.
